students/dave_arasim/session04/mailroom2.py

```diff
@@ -0,0 +1,222 @@
+
+#############################################################################################
+#Function section
+
+def main():
+    global this_os
+    global quit_option
+
+    donor_dict = defaultdict(list)
+    build_donor_dict(donor_dict)    #Build the Donor Dictionary from a text file
+
+    switch_dict = {1: send_thank_you, 2: create_report, 3: send_all_thanks, 4: quit_option_true}
+
+    while not(quit_option):
+        if this_os == 'Windows':
+            os.system('clear')
+        else:
+            os.system('cls')
+
+        print ('*'*10,'  Main Menu  ','*'*10)
+        print()
+        print('1) Send a Thank You to a single donor')
+        print('2) Create a Report')
+        print('3) Send Thank You letters to all donors')
+        print('4) Quit')
+        print()
+        print('Choose an option: ', end='')
+
+        user_choice = input()
+        print()
+
+        try:
+            user_choice = int(user_choice)
+        except ValueError:
+            user_choice = ''
+
+        if user_choice == '':
+            print('Sorry, but that is not a number')
+        else:
+            try:
+                switch_dict.get(user_choice)(donor_dict)
+            except TypeError:
+                print('That option: ', user_choice, ' is out of range', sep='')
+
+        print()
+
+        if quit_option:
+            print('Quitting process...')
+        else:
+            print('<cr> to continue... ', end='')
+            cont_input = input()  #cont_input for pause (variable not used)
+
+    build_donor_text(donor_dict)  #Export the Donor Dictionary back to a text file
+
+def quit_option_true(donor_dict):
+    '''Set the quit_option to true, so function can be selected from dict switch'''
+    global quit_option
+    quit_option = True
+
+def build_donor_dict(donor_dict):
+    '''Build Donor Dictionary by extracting values from a text file'''
+    try:
+        with open('donor_dict.txt', 'r') as dictfile:
+            while True:
+                donor_dict_line = dictfile.readline()
+                if not donor_dict_line: break
+
+                donor_dict_line = donor_dict_line.strip()
+                donor_dict_name = donor_dict_line.split(':')[0]
+                donor_dict_amts = donor_dict_line.split(':')[1].split(',')
+
+                for amts in donor_dict_amts:
+                    donor_dict[donor_dict_name].append(float(amts))
+
+        dictfile.close()
+    except FileNotFoundError:
+        print()
+        print('Donor database in donor_dict.txt not found.  Starting with new, empty database.')
+        print()
+
+def build_donor_text(donor_dict):
+    '''Export the Donor Dictionary date back to a text file'''
+    with open('donor_dict.txt', 'w') as dictfile:
+        for this_name, this_amts in donor_dict.items():
+            amts_str  = ','.join(str(x) for x in this_amts)
+            this_line = this_name + ':' + amts_str + '\n'
+
+            dictfile.write(this_line)
+
+    dictfile.close()
```

Comment:
Great to see you went extra miles with saving updated data back to your data file, which isn't a requirement at all!